llvm/lib/Target/NVPTX/NVPTXLowerUnreachable.cpp

```diff
@@ -99,6 +105,24 @@ char NVPTXLowerUnreachable::ID = 1;
 INITIALIZE_PASS(NVPTXLowerUnreachable, "nvptx-lower-unreachable",
                 "Lower Unreachable", false, false)
 
+StringRef NVPTXLowerUnreachable::getPassName() const {
+  return "add an exit instruction before every unreachable";
+}
+
+// =============================================================================
+// Returns whether a `trap` intrinsic should be emitted before I.
+//
+// This is a copy of the logic in SelectionDAGBuilder::visitUnreachable().
+// =============================================================================
+bool NVPTXLowerUnreachable::shouldEmitTrap(const UnreachableInst &I) const {
+  if (!TrapUnreachable)
```

Comment:
Nit: `if (!(TrapUnreachable || NoTrapAfterNoreturn))`

Reply:
I'm not sure I understand. `NoTrapAfterNoreturn` has no effect if `TrapUnreachable == false`.

Reply:
I'm suggesting collapsing two early exits into one. Sorry, I should've marked that the comment applies to more than one line. I'm still figuring out pull request review mechanics.

Reply:
```
if (!(TrapUnreachable && NoTrapAfterNoreturn))
  return TrapUnreachable;
```
... seems unnecessarily complex to save two LOCs, no?

Reply:
LLVM style tends to favor brevity, so if multiple exits do not buy us anything in terms of functionality or readability, fewer lines wins. This way you can see more code without having to scroll up/down, which does help with larger functions.

In this case it's fine either way as the function is small.